addons/survey/models/survey_survey.py

```diff
@@ -383,12 +384,13 @@ def _compute_session_available(self):
 
     @api.depends_context('uid')
     def _compute_allowed_survey_types(self):
-        self.allowed_survey_types = [
-            'survey',
-            'live_session',
-            'assessment',
-            'custom',
-        ] if self.env.user.has_group('survey.group_survey_user') else False
+        """Assign static mapping of allowed survey types and respective icons depending on users groups."""
+        self.allowed_survey_types = {
```

Comment:
Name could be updated, but don't think we are going to migrate just for that, so useless comment.